build.ps1

```diff
@@ -759,7 +759,7 @@ function Build-ICU($Arch) {
     }
   }
 
-  if ($Arch -eq $ArchARM64) {
+  if ($Arch -eq $ArchARM64 -and $HostArch -ne $ArchARM64) {
```

Comment:
Since the body of the if explicitly mentions `$ArchX64`, it should probably appear in the condition? `$HostArch -eq $ArchX64`?